miden-lib/asm/miden/kernels/tx/note.masm

```diff
@@ -118,17 +118,23 @@ end
 #! 2. Loading the note script root onto the stack.
 #!
 #! Stack: []
-#! Output: [NSR]
+#! Output: [NSR, NOTE_ARGS]
 #!
 #! - NSR is the note script root of the note currently being executed.
+#! - NOTE_ARGS are the note args of the note currently being executed.
 export.prepare_note
     # convert the index of the consumed note being executed to a pointer and store in memory
     exec.memory::get_current_consumed_note_ptr
     # => [note_ptr]
 
+    # get note args for the note 
+    # TODO: this should only be called if there are note_args
+    dup exec.memory::get_consumed_note_args movup.4
+    # => [note_ptr, NOTE_ARGS]
```

Comment:
I think it is fine to call this even if there are note args. In this case, we can assume that note args are set to `[ZERO; 4]`.

Also, we should probably update the comments in the `main` MASM procedure (where `prepare_note` is called).